src/geojson/geojson.js

```diff
@@ -0,0 +1,452 @@
+            layer.querySelector("layer-").setAttribute("label", "Layer");
+        }
+    }
+    let point = "<map-point></map-point>";
+    point = parser.parseFromString(point, "text/html");
+
+    let multiPoint = "<map-multipoint><map-coordinates></map-coordinates></map-multipoint>";
+    multiPoint = parser.parseFromString(multiPoint, "text/html");
+
+    let linestring = "<map-linestring><map-coordinates></map-coordinates></map-linestring>";
+    linestring = parser.parseFromString(linestring, "text/html");
+
+    let multilinestring = "<map-multilinestring></map-multilinestring>";
+    multilinestring = parser.parseFromString(multilinestring, "text/html");
+
+    let polygon = "<map-polygon></map-polygon>";
+    polygon = parser.parseFromString(polygon, "text/html");
+
+    let multiPolygon = "<map-multipolygon></map-multipolygon>";
+    multiPolygon = parser.parseFromString(multiPolygon, "text/html");
+
+    let geometrycollection = "<map-geometrycollection></map-geometrycollection>";
+    geometrycollection = parser.parseFromString(geometrycollection, "text/html");
+
+    let feature = "<map-feature><map-featurecaption>" + layer.querySelector("layer-").getAttribute('label') + "</map-featurecaption><map-geometry></map-geometry><map-properties></map-properties></map-feature>";
+    feature = parser.parseFromString(feature, "text/html");
+
+    // Template to add coordinates to Geometries
+    let coords = "<map-coordinates></map-coordinates>";
+    coords = parser.parseFromString(coords, "text/html");
+    
+    //console.log(layer);
+    if (jsonType === "FEATURECOLLECTION") {
+        let features = json.features;
+        //console.log("Features length - " + features.length);
+        for (let l=0;l<features.length;l++) {
+            geojson2mapml(features[l], properties, geometryFunction, layer);
+        }
+    } else if (jsonType === "FEATURE") {
+
+        let clone_feature = feature.cloneNode(true);
+        let curr_feature = clone_feature.querySelector('map-feature');
+
+        // Setting Properties
+        let p;
+        // if properties function is passed
+        if (typeof properties === "function") {
+            p = properties(json.properties);
+        } else if (typeof properties === "string") { // if properties string is passed
+            p = parser.parseFromString(properties, "text/xml").childNodes[0];
+        } else { // If no properties function or string is passed
+            p = properties2Table(json.properties);
+        }
+        
+        //console.log(p);
+        curr_feature.querySelector('map-properties').appendChild(p);
+
+        // Setting map-geometry
+        let g = geojson2mapml(json.geometry, properties, geometryFunction, layer);
+        if (typeof geometryFunction === "function") {
+            curr_feature.querySelector('map-geometry').appendChild(geometryFunction(g, json));
+        } else {
+            curr_feature.querySelector('map-geometry').appendChild(g);
+        }

```

Comment:
I'm confused why the "g" is a parameter of a passed geometryFunction -- shouldn't it just be the json.geometry that is passed, with the expectation that a \<mapml-geometry> is returned?